### DISCUSSION TITLE
two suggestions

### DISCUSSION THREAD

**Discussion by baucd** [y-young/nazurin#15] [General]
Firstly I would like to thank the author for this wonderful tool! I spent quite some time trying to host it on heroku and found two issues (branch master code).
1. the **start_webhook** method according to the [python-telegram-bot](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Webhooks) should have **webhook_url** as input while in the code it is missing. And I found no luck setup the bot without **webhook_url**.
2. It is better to put some deley between **start_webhook** and **setWebhook** otherwise error **telegram.error.RetryAfter: Flood control exceeded. Retry in 1.0 seconds** will block the following code from executing, especially the storage.load() line. The consequence is that the **mega** will not be connected.

These were what I experienced during the setup process and now it is working like a charm! Thanks

**Comment by y-young** [comment]
Thanks for your suggestions!
According to the revision history of PTB, they added this parameter about 2 weeks ago when I was planning the v2 release, so I didn't noticed. The v2 version of Nazurin will switch to aiogram, an asynchronous alternative of PTB, so this issue won't bother in v2. Also, in v2 we placed `storage.load()` before `setWebhook`, so this shouldn't be a problem too.
In a word, these issues should be fixed in the next version, I haven't decided whether to implement more new features before stable release, but I'll probably release a beta version recently.
The v2 version will come with better performance and exciting features, so please have a try if you're interested.

**Reply by baucd** [reply]
Sure! Looking forward to the release
